src/EditorFeatures/CSharpTest/MakeStructFieldsWritable/MakeStructFieldsWritableTests.cs

```diff
@@ -0,0 +1,597 @@
+    expected: Diagnostic(IDEDiagnosticIds.MakeStructFieldsWritable));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructFieldsWritable)]
+        public async Task SingleReadonlyField_InClass()
+        {
+            await TestDiagnosticMissingAsync(
+@"class [|MyClass|]
+{
+    public readonly int Value;
+
+    public MyClass(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyClass(5);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructFieldsWritable)]
+        public async Task StructWithoutField()
+        {
+            await TestDiagnosticMissingAsync(
+@"struct [|MyStruct|]
+{
+    public void Test()
+    {
+        this = new MyStruct();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructFieldsWritable)]
+        public async Task SingleProperty_ThisAssigmentInMethod()
+        {
+            await TestDiagnosticMissingAsync(
+@"struct [|MyStruct|]
+{
+    public int Value { get; set; }
+
+    public MyStruct(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyStruct(5);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructFieldsWritable)]
+        public async Task SingleGetterProperty_ThisAssigmentInMethod()
+        {
+            await TestDiagnosticMissingAsync(
+@"struct [|MyStruct|]
+{
+    public int Value { get; }
+
+    public MyStruct(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyStruct(5);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructFieldsWritable)]
+        public async Task MultipleStructDeclaration_SingleReadonlyField_ThisAssigmentInMethod()
+        {
+            await TestInRegularAndScriptAsync(
+@"struct MyStruct
+{
+    public readonly int Value;
+
+    public MyStruct(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyStruct(5);
+    }
+}
+
+struct [|MyStruct2|]
+{
+    public readonly int Value;
+
+    public MyStruct2(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyStruct2(5);
+    }
+}",
+@"struct MyStruct
+{
+    public readonly int Value;
+
+    public MyStruct(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyStruct(5);
+    }
+}
+
+struct MyStruct2
+{
+    public int Value;
+
+    public MyStruct2(int value)
+    {
+        Value = value;
+    }
+
+    public void Test()
+    {
+        this = new MyStruct2(5);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructFieldsWritable)]
+        public async Task MultipleStructDeclaration_SingleReadonlyField_ThisAssigmentInMethod_ShouldNotReprotDiagnostic()
```

Comment:
Nit: typo `ShouldNotReprotDiagnostic`

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/32063/commits/63e0d552020a3e64067db485a2193f35260b19b5](https://github.com/dotnet/roslyn/pull/32063/commits/63e0d552020a3e64067db485a2193f35260b19b5)